### PR TITLE
double quote name when part of the email address

### LINKED ISSUE #3230: Weird characters in names cause email failure
It looks like certain characters in a user's name can cause emails to fail bring sent. Specifically, Columbia reports that a user with an apostrophe in their name caused an email to fail as part of a class student email list.

Reported by Columbia Splash.

### PR DESCRIPTION
**Pull request by hwatheod**
This fixes #3230 by ensuring that names with special characters like commas and periods are processed correctly. The ESPUser.email_sendto_address now encloses the name in double quotes, and escapes existing double quotes with a backslash.

Also, change all the places where we manually build email recipients to use the ESPUser.email_sendto_address function instead.

NOTE TO REVIEWERS: **Please double check** that my changes from the old code to the ESPUser.email_sendto_address function are correct. Particularly note that the order of parameters for ESPUser.email_sendto_address is email **first** then name, whereas the original code was in the opposite order.

These changes will probably be easier to review if you view the diff in "Unified" mode.

### CODE REVIEW

**Review by willgearty — APPROVED** [review]
I double checked all of the changes and it all checks out, including the order of the arguments. Therefore, I'm approving this, but I'll also add that we could probably use the `get_email_sendto_address()` method to clean up the code in several places where we're working with an actual ESPUser object (we'll need to keep using the static method where we are using the director_email or something similar). It's up to you @hwatheod.

**Comment by hwatheod** [comment]
@willgearty Good call on using `get_email_sendto_address()`.  Made those changes. Can you verify again?

File: esp/esp/dbmail/receivers/classlist.py [hunk 1]
```diff
@@ -25,21 +25,18 @@ def process_nomailman(self, user, class_id, user_type):
...
-        self.recipients = ['%s Directors <%s>' % (program.niceName(), program.director_email)]
+        self.recipients = [ESPUser.email_sendto_address(program.director_email, '%s Directors' % (program.niceName()))]
 
         user_type = user_type.strip().lower()
 
         if user_type in ('teachers','class'):
-            self.recipients += ['%s %s <%s>' % (user.first_name,
-                                                user.last_name,
-                                                user.email)
+            self.recipients += [ESPUser.email_sendto_address(user.email, '%s %s' % (user.first_name, user.last_name))
```
Comment (willgearty) [inline esp/esp/dbmail/receivers/classlist.py:33]:
```suggestion
            self.recipients += [user.get_email_sendto_address()
```

File: esp/esp/dbmail/receivers/classlist.py [hunk 2]
```diff
@@ -25,21 +25,18 @@ def process_nomailman(self, user, class_id, user_type):
...
                                 for user in cls.get_teachers()     ]
 
         if user_type in ('students','class'):
             for section in sections:
-                self.recipients += ['%s %s <%s>' % (user.first_name,
-                                                    user.last_name,
-                                                    user.email)
+                self.recipients += [ESPUser.email_sendto_address(user.email, '%s %s' % (user.first_name,
+                                                                                        user.last_name))
```
Comment (willgearty) [inline esp/esp/dbmail/receivers/classlist.py:39]:
```suggestion
                self.recipients += [user.get_email_sendto_address()
```

File: esp/esp/dbmail/receivers/sectionlist.py [hunk 3]
```diff
@@ -29,20 +30,18 @@ def process_nomailman(self, user, class_id, section_num, user_type):
...
+        self.recipients = [ESPUser.email_sendto_address(program.director_email, '%s Directors' % (program.niceName()))]
 
         user_type = user_type.strip().lower()
 
         if user_type in ('teachers','class'):
-            self.recipients += ['%s %s <%s>' % (user.first_name,
-                                                user.last_name,
-                                                user.email)
+            self.recipients += [ESPUser.email_sendto_address(user.email, '%s %s' % (user.first_name,
+                                                                                    user.last_name))
```
Comment (willgearty) [inline esp/esp/dbmail/receivers/sectionlist.py:39]:
```suggestion
            self.recipients += [user.get_email_sendto_address()
```

File: esp/esp/dbmail/receivers/sectionlist.py [hunk 4]
```diff
@@ -29,20 +30,18 @@ def process_nomailman(self, user, class_id, section_num, user_type):
...
                                 for user in section.parent_class.get_teachers()     ]
 
         if user_type in ('students','class'):
-            self.recipients += ['%s %s <%s>' % (user.first_name,
-                                                user.last_name,
-                                                user.email)
+            self.recipients += [ESPUser.email_sendto_address(user.email, '%s %s' % (user.first_name,
+                                                                                    user.last_name))
```
Comment (willgearty) [inline esp/esp/dbmail/receivers/sectionlist.py:44]:
```suggestion
            self.recipients += [user.get_email_sendto_address()
```

File: esp/esp/program/modules/handlers/teacherclassregmodule.py [hunk 5]
```diff
@@ -445,13 +445,13 @@ def cancelrequest(self, request, tl, one, two, module, extra, prog):
             #Send email to all teachers confirming cancellation request
             email_contents = render_to_string('program/modules/teacherclassregmodule/cancelrequest.txt', email_context)
             for teacher in cls.get_teachers():
-                email_to = ['%s <%s>' % (teacher.name(), teacher.email)]
+                email_to = [ESPUser.email_sendto_address(teacher.email, teacher.name())]
```
Comment (willgearty) [inline esp/esp/program/modules/handlers/teacherclassregmodule.py:448]:
```suggestion
                email_to = [teacher.get_email_sendto_address()]
```

File: esp/esp/program/modules/handlers/teachereventsmodule.py [hunk 6]
```diff
@@ -144,7 +144,7 @@ def event_signup(self, request, tl, one, two, module, extra, prog):
                         send_mail('['+self.program.niceName()+'] Teacher Interview for ' + request.user.first_name + ' ' + request.user.last_name + ': ' + event_name, \
                               """Teacher Interview Registration Notification\n--------------------------------- \n\nTeacher: %s %s\n\nTime: %s\n\n""" % \
                               (request.user.first_name, request.user.last_name, event_name) , \
-                              ('%s <%s>' % (request.user.first_name + ' ' + request.user.last_name, request.user.email,)), \
+                              (ESPUser.email_sendto_address(request.user.email, request.user.first_name + ' ' + request.user.last_name)), \
```
Comment (willgearty) [inline esp/esp/program/modules/handlers/teachereventsmodule.py:147]:
```suggestion
                              (request.user.get_email_sendto_address()), \
```

**Review by willgearty — CHANGES_REQUESTED** [review]
Found a few stragglers

File: esp/useful_scripts/finaid_autoapproved.py [hunk 7]
```diff
@@ -8,7 +8,7 @@
 #
 
 from script_setup import *
-
+from esp.users.models import ESPUser
```
Comment (willgearty) [inline esp/useful_scripts/finaid_autoapproved.py:11]: Don't need this import anymore

**Review by willgearty — APPROVED** [review]
Looks good to me, thanks @hwatheod!

**Pull request merged**